spec.bs

```diff
@@ -810,6 +810,83 @@ To <dfn>compute protocol matches a special scheme flag</dfn> given a [=construct
   1. If the result of running [=protocol component matches a special scheme=] given |protocol component| is true, then set |parser|'s [=constructor string parser/protocol matches a special scheme flag=] to true.
 </div>
 
+<h2 id=urlpatternlist-class>The {{URLPatternList}} class </h2>
+
+<xmp class="idl">
+[Exposed=(Window,Worker)]
+interface URLPatternList {
+  constructor(sequence<URLPatternListEntry> entries);
+
+  USVString? test(optional URLPatternInput input = {}, optional USVString baseURL);
+
+  URLPatternListResult? exec(optional URLPatternInput input = {}, optional USVString baseURL);
+};
+
+dictionary URLPatternListEntry {
+  required (USVString or URLPattern) pattern;
+  required USVString key;
```

Comment:
Can you explain the "key" a bit more?  It feels like its not really functioning as a key here and is instead some extra data associated with the pattern.  It seems the pattern is more like the "key" of the entry to me.

Also, why a string for the extra data?  Why not an `any` so you can store any javascript object?

Reply:
No real reason. I think we may even be able to get rid of this "key" or metadata thing entirely. Instead of returning the `"key"` from the matcher, we can return the `URLPattern` object that matched. Users can associate metadata with a pattern using a `Map` or `WeakMap`. That way the input can just be `sequence<URLPattern>`.